### PR TITLE
feat : upgraded the app with consensus module to SDK v0.50.0

### PR DESCRIPTION
**Pull request by ruthishvitwit**

### CODE REVIEW

File: x/evm/module.go [hunk 1]
```diff
@@ -19,14 +21,17 @@ import (
 )
 
 var (
-	_ module.AppModule      = AppModule{}
+	// _ module.AppModule      = AppModule{}
```
Comment (byte-bandit) [inline x/evm/module.go:24]: Can this be removed?

File: x/evm/keeper/smart_contract_deployment.go [hunk 2]
```diff
@@ -57,15 +58,15 @@ func (k Keeper) DeleteSmartContractDeploymentByContractID(ctx sdk.Context, smart
...
+	logger := k.Logger(ctx).With("smart-contract-id", smartContractID, "chain-reference-id", chainReferenceID, "new-smart-contract-status", status)
 	v, key := k.getSmartContractDeploymentByContractID(ctx, smartContractID, chainReferenceID)
 	if key == nil {
 		return keeperutil.ErrNotFound
 	}
 
 	v.Status = status
 	if err := keeperutil.Save(k.provideSmartContractDeploymentStore(ctx), k.cdc, key, v); err != nil {
-		logger.WithError(err).Error("failed to update smart contract deployment record")
+		logger.With(err).Error("failed to update smart contract deployment record")
```
Comment (byte-bandit) [inline x/evm/keeper/smart_contract_deployment.go:69]: Make sure to use `WithError` for logging errors please.

File: x/evm/keeper/smart_contract_deployment.go [hunk 3]
```diff
@@ -57,15 +58,15 @@ func (k Keeper) DeleteSmartContractDeploymentByContractID(ctx sdk.Context, smart
 }
 
 func (k Keeper) SetSmartContractDeploymentStatusByContractID(ctx sdk.Context, smartContractID uint64, chainReferenceID string, status types.SmartContractDeployment_Status) error {
-	logger := k.Logger(ctx).WithFields("smart-contract-id", smartContractID, "chain-reference-id", chainReferenceID, "new-smart-contract-status", status)
+	logger := k.Logger(ctx).With("smart-contract-id", smartContractID, "chain-reference-id", chainReferenceID, "new-smart-contract-status", status)
```
Comment (byte-bandit) [inline x/evm/keeper/smart_contract_deployment.go:61]: Make sure to use `WithFields` for setting logging fields. It returns a different type than the default `With()` and we want that.

**Review by byte-bandit — CHANGES_REQUESTED** [review]

File: x/evm/keeper/smart_contract_deployment.go (hunk 3 above)
Comment (ruthishvitwit) [reply x/evm/keeper/smart_contract_deployment.go:61]: @byte-bandit As we have coverted to cosmossdk.io/log ...there are no WithFields function in logger...should i just use the with??

Comment (byte-bandit) [reply x/evm/keeper/smart_contract_deployment.go:61]: Use the `liblog` package for this, it wraps the underlaying logger.

**Comment by aleem1314** [comment]
Check #1036 PR review comments. Apply relevant changes here too.

**Comment by byte-bandit** [comment]
Also https://github.com/palomachain/paloma/pull/1044

**Review by aleem1314 — CHANGES_REQUESTED** [review]
Still there are few things need to be updated.
Check https://github.com/palomachain/paloma/pull/1036#pullrequestreview-1763965986

File: x/valset/module.go [hunk 4]
```diff
@@ -19,7 +19,7 @@ import (
 )
 
 var (
-	_ module.AppModule      = AppModule{}
+	// _ module.AppModule      = AppModule{}
```
Comment (aleem1314) [inline x/valset/module.go:22]: revert this change

File: x/consensus/keeper/keeper.go [hunk 5]
```diff
@@ -29,16 +35,12 @@ type (
 
 func NewKeeper(
 	cdc codec.BinaryCodec,
-	storeKey,
+	storeKey store.KVStoreService,
 	memKey storetypes.StoreKey,
```
Comment (aleem1314) [inline x/consensus/keeper/keeper.go:39]: Why storeKey and memKey using different types?

File: x/consensus/module.go [hunk 6]
```diff
@@ -19,7 +20,7 @@ import (
 )
 
 var (
-	_ module.AppModule      = AppModule{}
+	// _ module.AppModule      = AppModule{}
```
Comment (aleem1314) [inline x/consensus/module.go:23]: Why this line commented?

File: x/consensus/keeper/consensus/batch_test.go [hunk 7]
```diff
@@ -43,7 +45,7 @@ func TestBatching(t *testing.T) {
 			QueueTypeName: "simple-message",
 			Sg:            sg,
 			Ider:          keeperutil.NewIDGenerator(sg, nil),
-			Cdc:           types.ModuleCdc,
+			Cdc:           nil,
```
Comment (byte-bandit) [inline x/consensus/keeper/consensus/batch_test.go:48]: I think this will create problems. The unpacker is used by different keeper methods.

**Review by byte-bandit — CHANGES_REQUESTED** [review]

File: x/consensus/keeper/consensus/consensus_test.go [hunk 8]
```diff
@@ -42,7 +44,7 @@ func TestConsensusQueueAllMethods(t *testing.T) {
 			QueueTypeName:         "simple-message",
 			Sg:                    sg,
 			Ider:                  keeperutil.NewIDGenerator(sg, nil),
-			Cdc:                   types.ModuleCdc,
+			Cdc:                   nil,
```
Comment (byte-bandit) [inline x/consensus/keeper/consensus/consensus_test.go:47]: Why is this `nil` now? I'm pretty sure that will break production and test code.

File: x/consensus/keeper/keeper.go [hunk 9]
```diff
@@ -54,10 +52,13 @@ func NewKeeper(
...
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	return k.Logger(sdkCtx).With("module", fmt.Sprintf("x/%s", types.ModuleName))
 }
 
-func (k Keeper) Store(ctx sdk.Context) sdk.KVStore {
-	return ctx.KVStore(k.storeKey)
+func (k Keeper) Store(ctx context.Context) storetypes.KVStore {
+	s := runtime.KVStoreAdapter(k.storeKey.OpenKVStore(ctx))
+
+	return prefix.NewStore(s, []byte("store"))
```
Comment (byte-bandit) [inline x/consensus/keeper/keeper.go:63]: Extract store key to types to remain in line with other modules.

File: x/consensus/types/message_add_evidence.go [hunk 10]
```diff
@@ -9,23 +9,10 @@ const TypeMsgAddEvidence = "add_evidence"
 
 var _ sdk.Msg = &MsgAddEvidence{}
 
-func (msg *MsgAddEvidence) Route() string {
-	return RouterKey
-}
-
-func (msg *MsgAddEvidence) Type() string {
-	return TypeMsgAddEvidence
-}
```
Comment (byte-bandit) [inline x/consensus/types/message_add_evidence.go:18]: We're using `Type()` in production code for some of these messages. I don't see that usage removed or changed in this PR. Why remove it in the first place? What's the new approach here?

**Review by byte-bandit — CHANGES_REQUESTED** [review]

File: x/consensus/keeper/consensus/consensus_test.go (hunk 8 above)
Comment (aleem1314) [reply x/consensus/keeper/consensus/consensus_test.go]: Resolved

File: x/consensus/keeper/keeper.go (hunk 9 above)
Comment (aleem1314) [reply x/consensus/keeper/keeper.go]: Fixed

File: x/consensus/types/message_add_evidence.go (hunk 10 above)
Comment (byte-bandit) [reply x/consensus/types/message_add_evidence.go:18]: For example here, https://github.com/palomachain/paloma/blob/2226e95be60c4972dff44afbbbe7b4cb0ec268c8/x/gravity/keeper/msg_server.go#L53 - but there's more places in the code.

Comment (aleem1314) [reply x/consensus/types/message_add_evidence.go]: Updated

File: x/consensus/keeper/concensus_keeper.go [hunk 11]
```diff
@@ -170,7 +178,7 @@ func (k Keeper) GetMessagesForRelaying(ctx sdk.Context, queueTypeName string, va
 	msgs = slice.Filter(msgs, func(msg types.QueuedSignedMessageI) bool {
 		var unpackedMsg evmtypes.TurnstoneMsg
 		if err := k.cdc.UnpackAny(msg.GetMsg(), &unpackedMsg); err != nil {
-			k.Logger(ctx).With("err", err).Error("Failed to unpack message")
+			k.Logger(sdkCtx).With("err", err).Error("Failed to unpack message")
```
Comment (byte-bandit) [inline x/consensus/keeper/concensus_keeper.go:181]:
```suggestion
			liblog.FromSDKLogger(k.Logger(sdkCtx)).WithError(err).Error("Failed to unpack message.")
```

File: x/consensus/keeper/concensus_keeper.go [hunk 12]
```diff
@@ -220,16 +229,17 @@ func (k Keeper) GetMessagesForAttesting(ctx sdk.Context, queueTypeName string, v
...
 
 // GetMessagesFromQueue gets N messages from the queue.
-func (k Keeper) GetMessagesFromQueue(ctx sdk.Context, queueTypeName string, n int) (msgs []types.QueuedSignedMessageI, err error) {
-	cq, err := k.getConsensusQueue(ctx, queueTypeName)
+func (k Keeper) GetMessagesFromQueue(ctx context.Context, queueTypeName string, n int) (msgs []types.QueuedSignedMessageI, err error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	cq, err := k.getConsensusQueue(sdkCtx, queueTypeName)
 	if err != nil {
-		k.Logger(ctx).Error("error while getting consensus queue", "err", err)
+		k.Logger(sdkCtx).Error("error while getting consensus queue", "err", err)
```
Comment (byte-bandit) [inline x/consensus/keeper/concensus_keeper.go:236]:
```suggestion
		liblog.FromSDKLogger(k.Logger(sdkCtx)).WithError(err).Error("error while getting consensus queue.")
```

File: x/consensus/keeper/concensus_keeper.go [hunk 13]
```diff
@@ -220,16 +229,17 @@ func (k Keeper) GetMessagesForAttesting(ctx sdk.Context, queueTypeName string, v
...
 		return nil, err
 	}
-	msgs, err = cq.GetAll(ctx)
+	msgs, err = cq.GetAll(sdkCtx)
 
 	if err != nil {
-		k.Logger(ctx).Error("error while getting all messages from queue", "err", err)
+		k.Logger(sdkCtx).Error("error while getting all messages from queue", "err", err)
```
Comment (byte-bandit) [inline x/consensus/keeper/concensus_keeper.go:242]:
```suggestion
		liblog.FromSDKLogger(k.Logger(sdkCtx)).WithError(err).Error("error while getting all messages from queue")
```

File: x/consensus/keeper/concensus_keeper.go [hunk 14]
```diff
@@ -240,32 +250,34 @@ func (k Keeper) GetMessagesFromQueue(ctx sdk.Context, queueTypeName string, n in
...
 }
 
-func (k Keeper) DeleteJob(ctx sdk.Context, queueTypeName string, id uint64) (err error) {
-	cq, err := k.getConsensusQueue(ctx, queueTypeName)
+func (k Keeper) DeleteJob(ctx context.Context, queueTypeName string, id uint64) (err error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	cq, err := k.getConsensusQueue(sdkCtx, queueTypeName)
 	if err != nil {
-		k.Logger(ctx).Error("error while getting consensus queue", "err", err)
+		k.Logger(sdkCtx).Error("error while getting consensus queue", "err", err)
```
Comment (byte-bandit) [inline x/consensus/keeper/concensus_keeper.go:257]:
```suggestion
		liblog.FromSDKLogger(k.Logger(sdkCtx)).WithError(err).Error("error while getting consensus queue")
```

File: x/consensus/keeper/concensus_keeper.go [hunk 15]
```diff
@@ -410,14 +428,14 @@ func (k Keeper) SetMessagePublicAccessData(
 		ValAddress: valAddr,
 		Data:       msg.GetData(),
 	}
-	err = cq.SetPublicAccessData(ctx, msg.GetMessageID(), payload)
+	err = cq.SetPublicAccessData(sdkCtx, msg.GetMessageID(), payload)
 	if err != nil {
-		k.Logger(ctx).Error("error while adding message public access data", "err", err)
+		k.Logger(sdkCtx).Error("error while adding message public access data", "err", err)
```
Comment (byte-bandit) [inline x/consensus/keeper/concensus_keeper.go:433]: Replace with liblog usage as above for other touched occurrences in this file and others.

Comment (aleem1314) [reply x/consensus/keeper/concensus_keeper.go]: updated

File: x/consensus/keeper/concensus_keeper.go [hunk 16]
```diff
@@ -341,37 +350,33 @@ func (k Keeper) AddMessageSignature(
...
 			)
 
-			k.Logger(ctx).Info("added message signature",
-				"message-id", msg.GetId(),
-				"queue-type-name", msg.GetQueueTypeName(),
-				"signed-by-address", msg.GetSignedByAddress(),
-				"chain-type", chainType,
-				"chain-reference-id", chainReferenceID,
-			)
+			liblog.FromSDKLogger(k.Logger(sdkCtx)).WithFields(msg.GetId(), msg.GetQueueTypeName(),
```
Comment (byte-bandit) [inline x/consensus/keeper/concensus_keeper.go:353]: Use proper kv pairs for `WithFields()`

```
"queue-type-name", msg.GetQueueTypeName(),
				"signed-by-address", msg.GetSignedByAddress(),
				"chain-type", chainType,
				"chain-reference-id", chainReferenceID,
```

File: x/consensus/keeper/concensus_keeper.go [hunk 17]
```diff
@@ -410,30 +411,26 @@ func (k Keeper) SetMessagePublicAccessData(
...
 
 	chainType, chainReferenceID := cq.ChainInfo()
-	k.Logger(ctx).Info("added message public access data",
-		"message-id", msg.GetMessageID(),
-		"queue-type-name", msg.GetQueueTypeName(),
-		"chain-type", chainType,
-		"chain-reference-id", chainReferenceID,
-		"public-access-data", hexutil.Encode(payload.Data),
-	)
+	liblog.FromSDKLogger(k.Logger(sdkCtx)).WithFields(msg.GetMessageID(),
```
Comment (byte-bandit) [inline x/consensus/keeper/concensus_keeper.go:421]: here as well

Comment (byte-bandit) [reply x/consensus/keeper/concensus_keeper.go:421]: actually everywhere you use it.

**Review by byte-bandit — COMMENTED** [review]
Only one small thing from my end.

**Review by byte-bandit — APPROVED** [review]
lgtm, final review to @aleem1314

**Review by aleem1314 — APPROVED** [review]
LGTM

**Comment by deepan95dev** [comment]
@alexanderbez  if everything looks good in this module, lets merge with v0.50.0-upgrade branch. so that we will update other modules accordingly which uses consensus module.

**Comment by taariq** [comment]
@byte-bandit do we not want checks to be passing before we merge?

**Comment by byte-bandit** [comment]
@taariq I don't think that's realistic. Changing each module separately is going to break all tests and CI. It's not great, but not horrible. We definitely want a full green CI pipeline before merging back to master though, with updated and expanded test coverage.

**Comment by byte-bandit** [comment]
@deepan95dev I disabled the need for green CI for now, but we need a cleaner commit history.

I suggest to go over the the 50 pending commits, rebase and fixup where useful. Please use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). We require signatures as well: https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification

Once that's done, feel free to mege.

**Comment by byte-bandit** [comment]
@deepan95dev Actually, I enabled the linter again. There's no reason why we should not keep it green. Please make sure it passes as well.

**Comment by aleem1314** [comment]
> @deepan95dev Actually, I enabled the linter again. There's no reason why we should not keep it green. Please make sure it passes as well.

Hey @byte-bandit , We need to fix the `go.mod` file before running the linter. After updating all modules, we'll create separate pull requests to address test and lint issues.

**Comment by byte-bandit** [comment]
@aleem1314 Is there a way we can keep the `go.mod` alive for these PRs? We can reduce the noise of auto generated files and whitespace changes by enforcing linter usage.

**Comment by aleem1314** [comment]
> @aleem1314 Is there a way we can keep the `go.mod` alive for these PRs? We can reduce the noise of auto generated files and whitespace changes by enforcing linter usage.

> I don't think it is possible, because few modules are pointing to SDK v0.47.x and few modules pointing to v0.50.x. `go mod` updates always failing because of this.

**Comment by byte-bandit** [comment]
> > @aleem1314 Is there a way we can keep the `go.mod` alive for these PRs? We can reduce the noise of auto generated files and whitespace changes by enforcing linter usage.
> 
> I don't think it is possible, because few modules are pointing to SDK v0.47.x and few modules pointing to v0.50.x. `go mod` updates always failing because of this.

Okay, let's keep it rolling. Need for linters removed for `v0.50.0-upgrade` branch.

**Comment by deepan95dev** [comment]
@byte-bandit rebase has been done. Please approve to merge this PR

**Review by taariq — APPROVED** [review]
LGTM!

**Comment by taariq** [comment]
@aleem1314 Need your final review here as well to merge. Thanks!

**Review by aleem1314 — APPROVED** [review]

**Pull request merged**